homeassistant/components/fan/services.yaml

```diff
@@ -120,8 +120,76 @@ xiaomi_miio_set_led_brightness:
   description: Set the led brightness.
   fields:
     entity_id:
-      description: Name of the air purifier entity.
-      example: 'fan.xiaomi_air_purifier'
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
     brightness:
       description: Brightness (0 = Bright, 1 = Dim, 2 = Off)
       example: 1
+
+xiaomi_miio_set_auto_detect_on:
+  description: Turn the auto detect on.
+  fields:
+    entity_id:
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
+    brightness:
+      description: Brightness (0 = Bright, 1 = Dim, 2 = Off)
+      example: 1
+
+xiaomi_miio_set_auto_detect_off:
+  description: Turn the auto detect off.
+  fields:
+    entity_id:
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
+
+xiaomi_miio_set_learn_mode_on:
+  description: Turn the learn mode on.
+  fields:
+    entity_id:
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
+
+xiaomi_miio_set_learn_mode_off:
+  description: Turn the learn mode off.
+  fields:
+    entity_id:
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
+
+xiaomi_miio_set_volume:
+  description: Set the sound volume.
+  fields:
+    entity_id:
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
+    volume:
+      description: Volume, between 0 and 100.
+      example: 50
+
+xiaomi_miio_reset_filter:
+  description: Reset the filter lifetime and usage.
+  fields:
+    entity_id:
+      description: Name of the xiaomi miio entity.
+      example: 'fan.xiaomi_miio_device'
+
+xiaomi_miio_set_extra_features:
+  description: Set the extra features.
```

Comment:
This could be more elaborate, what are the extra features and what are the known values below?